### ISSUE TITLE
archiving each sample individually

### ISSUE THREAD

**Issue by nsheff** [databio/refgenieserver#84]
right now we just run `archive` and it archives the whole thing at once...

is it possible to split this into archive tasks so that I can farm these out as sample jobs (per asset) instead of a single job for the whole shebang?

**Comment by stolarczyk** [comment]
yes, I think it is possible to parallelize this since the archives are totally independent of each other and `RefGenConf` supports locking and race-free writing.  The problem is, we initialize this job based on refgenie config file, which would need to be transformed into PEP and we could write this as a pypiper pipeline and submit using looper.

**Comment by nsheff** [comment]
It would be simpler to just:

- modifiy the `refgenieserver archive` command to accept a registry path (like `refgenieserver archive hg38/bowtie2_index`)
- write a sample-level archive piface (instead of the current project-level one)
- take the original pep as input and run the archiver

**Comment by stolarczyk** [comment]
> modifiy the refgenieserver archive command to accept a registry path (like refgenieserver archive hg38/bowtie2_index)

It does already

```
usage: refgenieserver archive [-h] [-c CONFIG] [-d]
                              [--genomes-desc GENOMES_DESC] [-f] [-r]
                              [asset-registry-paths [asset-registry-paths ...]]

prepare servable archives

positional arguments:
  asset-registry-paths  One or more registry path strings that identify
                        assets, e.g. hg38/fasta:tag

...
```

**Comment by nsheff** [comment]
well then :). that makes this even easier...

it will be much nicer to archive in parallel.

**Comment by stolarczyk** [comment]
I can try to set this up. Does it still live databio_genomes? or dbg?

**Comment by nsheff** [comment]
everything's a mess right now... give me an hour and I'll get things a bit more organized.

it will live in refgenie/server_deploy

**Comment by nsheff** [comment]
this is working fine, it turns out (after we solve all the issues with parallel writing to the config file...)
